File: templates/Uwp/_composition/MVVMLight/Project.MenuBar/ViewModels/ShellViewModel.cs
using System.Windows.Input;

using Param_RootNamespace.Helpers;
using Param_RootNamespace.Services;

using GalaSoft.MvvmLight;
using GalaSoft.MvvmLight.Command;

using Windows.UI.Xaml;
using Windows.UI.Xaml.Controls;

namespace Param_RootNamespace.ViewModels
{
    public class ShellViewModel : ViewModelBase
    {
        private ICommand _menuFileExitCommand;

        public ICommand MenuFileExitCommand => _menuFileExitCommand ?? (_menuFileExitCommand = new RelayCommand(OnMenuFileExit));

        public NavigationServiceEx NavigationService => ViewModelLocator.Current.NavigationService;

        public ShellViewModel()
        {
        }

        public void Initialize(Frame shellFrame, SplitView splitView, Frame rightFrame)
        {
            NavigationService.Frame = shellFrame;
            MenuNavigationHelper.Initialize(splitView, rightFrame);
        }

        private void OnMenuFileExit()
        {
            Application.Current.Exit();
        }
    }
}

File: templates/Uwp/_composition/MVVMLight/Project.MenuBar._VB/ViewModels/ShellViewModel.vb
Imports System.Windows.Input
Imports Param_RootNamespace.Helpers
Imports Param_RootNamespace.Services
Imports GalaSoft.MvvmLight
Imports GalaSoft.MvvmLight.Command
Imports Windows.UI.Xaml
Imports Windows.UI.Xaml.Controls

Namespace ViewModels
    Public Class ShellViewModel
        Inherits ViewModelBase

        Private _menuFileExitCommand As ICommand

        Public ReadOnly Property MenuFileExitCommand As ICommand
            Get
                If _menuFileExitCommand Is Nothing Then
                    _menuFileExitCommand = New RelayCommand(AddressOf OnMenuFileExit)
                End If

                Return _menuFileExitCommand
            End Get
        End Property

        Public ReadOnly Property NavigationService As NavigationServiceEx
            Get
                Return ViewModelLocator.Current.NavigationService
            End Get
        End Property

        Public Sub New()
        End Sub

        Public Sub Initialize(shellFrame As Frame, splitView As SplitView, rightFrame As Frame)
            NavigationService.Frame = shellFrame
            MenuNavigationHelper.Initialize(splitView, rightFrame)
        End Sub

        Private Sub OnMenuFileExit()
            Application.Current.[Exit]()
        End Sub
    End Class
End Namespace
